src/test/scala/com/microsoft/hyperspace/index/IndexManagerTests.scala

```diff
@@ -251,6 +251,90 @@ class IndexManagerTests extends HyperspaceSuite with SQLHelper {
     }
   }
 
+  test("Verify refresh-incremental (append-only) throws exception if no new files found.") {
+    withTempDir { testDir =>
+      val refreshTestLocation = testDir + "/refresh"
```

Comment:
I see many of these patterns now. Can you add the following to `HyperspaceSuite.scala`:
```scala
  protected def withTempPathAsString(f: String => Unit): Unit = {
    // The following is from SQLHelper.withTempPath with a modification to pass
    // String instead of File to "f". The reason this is copied instead of extending
    // SQLHelper is that some of the existing suites extend QueryTest and it causes
    // "inheriting conflicting members" issue.
    val path = Utils.createTempDir()
    path.delete()
    try f(path.toString) finally Utils.deleteRecursively(path)
  }
```
, and use it as
```scala
    withTempPathAsString { testPath =>
      withSQLConf(IndexConstants.REFRESH_APPEND_ENABLED -> "true") {
        // Setup. Create sample data and index.
        val indexConfig = IndexConfig(s"index", Seq("RGUID"), Seq("imprs"))
        import spark.implicits._
        SampleData.testData
          .toDF("Date", "RGUID", "Query", "imprs", "clicks")
          .limit(10)
          .write
          .parquet(testPath)
        val df = spark.read.parquet(testPath)
        hyperspace.createIndex(df, indexConfig)
        val ex = intercept[HyperspaceException] {
          hyperspace.refreshIndex(indexConfig.indexName)
        }
        assert(ex.msg.equals("Refresh aborted as no appended source data files found."))
      }
    }
```

```

Reply:
thanks @imback82 , this is nice!